unic/ucd/normal/src/decomposition_type.rs

```diff
@@ -12,228 +12,180 @@
+        Canonical {
+            abbr => Can,
+            long => Canonical,
+            human => "Canonical",
+        }
 
-/// Represents the Unicode character
-/// [*Decomposition_Type*](http://www.unicode.org/reports/tr44/#Decomposition_Type) property.
-///
-/// * <http://www.unicode.org/reports/tr44/#Character_Decomposition_Mappings>
-#[derive(Clone, Copy, Debug, Eq, Hash, PartialEq)]
-#[allow(missing_docs)]
-pub enum DecompositionType {
-    Canonical,
-    Compat,
-    Circle,
-    Final,
-    Font,
-    Fraction,
-    Initial,
-    Isolated,
-    Medial,
-    Narrow,
-    Nobreak,
-    None,
-    Small,
-    Square,
-    Sub,
-    Super,
-    Vertical,
-    Wide,
-}
-
+        /// Otherwise unspecified compatibility character
+        Compat {
+            abbr => Com,
+            long => Compat,
+            human => "Unspecified",
+        }
 
-impl CharProperty for DecompositionType {
-    fn prop_abbr_name() -> &'static str {
-        "dt"
-    }
+        /// Encircled form
+        Circle {
+            abbr => Enc,
+            long => Circle,
+            human => "Encircled",
+        }
 
-    fn prop_long_name() -> &'static str {
-        "Decomposition_Type"
-    }
+        /// Final presentation form (Arabic)
+        Final {
+            abbr => Fin,
+            long => Final,
+            human => "Arabic Final",
+        }
 
-    fn prop_human_name() -> &'static str {
-        "Decomposition Type"
-    }
-}
+        /// Font variant (for example, a blackletter form)
+        Font {
+            abbr => Font,
+            long => Font,
+            human => "Font Variant",
+        }
 
+        /// Vulgar fraction form
+        Fraction {
+            abbr => Fra,
+            long => Fraction,
+            human => "Vulgar Fraction",
+        }
 
-impl PartialCharProperty for DecompositionType {
-    fn of(ch: char) -> Option<Self> {
-        Self::of(ch)
-    }
-}
+        /// Initial presentation form (Arabic)
+        Initial {
+            abbr => Init,
+            long => Initial,
+            human => "Arabic Initial",
+        }
 
+        /// Isolated presentation form (Arabic)
+        Isolated {
+            abbr => Iso,
+            long => Isolated,
+            human => "Arabic Isolated",
+        }
 
-impl EnumeratedCharProperty for DecompositionType {
-    fn all_values() -> &'static [Self] {
-        Self::all_values()
-    }
+        /// Medial presentation form (Arabic)
+        Medial {
+            abbr => Med,
+            long => Medial,
+            human => "Arabic Medial",
+        }
 
-    fn abbr_name(&self) -> &'static str {
-        self.abbr_name()
-    }
+        /// Narrow (or hankaku) compatibility character
+        Narrow {
+            abbr => Nar,
+            long => Narrow,
+            human => "Narrow",
+        }
 
-    fn long_name(&self) -> &'static str {
-        self.long_name()
-    }
+        /// No-break version of a space or hyphen
+        NoBreak {
```

Comment:
Oops! I meant "does *not* provide much gain".

Yeah, totally agree with all the arguments here. But, also, similar to some other cases before, we are going to need a clear way to explain it as a generic overall rule.

For this, the fact that the spec and data files use `noBreak` is not enough to define the API based on. I mean, the more formal form, `PropertyAliases.txt`, is more important, and I want us to be able to describe the decisions based on that more formal data.

But agreed, this one term is in such a messy situation now. Unfortunately there's a tendency to not fix these things in the updates.

To be clear, I wish we can come up with a policy to justify it. I just can't think of any yet.